packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -193,12 +221,26 @@ class HttpRouteSpec extends RouteSpec {
       match: {
         prefix: prefixPath,
       },
+      timeout: this.timeout ? this.renderTimeout(this.timeout): undefined,
     };
     return {
       httpRouteSpec: this.protocol === Protocol.HTTP ? httpConfig : undefined,
       http2RouteSpec: this.protocol === Protocol.HTTP2 ? httpConfig : undefined,
     };
   }
+
+  private renderTimeout(timeout: HttpTimeout): CfnRoute.HttpTimeoutProperty {
```

Comment:
This method is duplicated between all 3 classes.

Let's use the magic of TypeScript's static typing here, and have a single `renderTimeout` function that takes an `HttpTimeout` as argument, and returns `CfnRoute.HttpTimeoutProperty`. I _believe_ that it will work from the other classes too, not only in `HttpRouteSpec`.

Reply:
ack, using a common helper method and removing the reused ones per class.